pkg/api/customization/catalog/template.go

```diff
@@ -17,17 +17,33 @@ import (
 )
 
 func TemplateFormatter(apiContext *types.APIContext, resource *types.RawResource) {
+	var prjCatalogName string
 	// version links
 	resource.Values["versionLinks"] = extractVersionLinks(apiContext, resource)
 
 	//icon
 	delete(resource.Values, "icon")
 	resource.Links["icon"] = apiContext.URLBuilder.Link("icon", resource)
 
-	//catalog link
-	catalogSchema := apiContext.Schemas.Schema(&managementschema.Version, client.CatalogType)
-	catalogName := strings.Split(resource.ID, "-")[0]
-	resource.Links["catalog"] = apiContext.URLBuilder.ResourceLinkByID(catalogSchema, catalogName)
+	val := resource.Values
+	if val[client.TemplateFieldCatalogID] != nil {
+		//catalog link
+		catalogSchema := apiContext.Schemas.Schema(&managementschema.Version, client.CatalogType)
+		catalogName := strings.Split(resource.ID, "-")[0]
+		resource.Links["catalog"] = apiContext.URLBuilder.ResourceLinkByID(catalogSchema, catalogName)
+	}
+
+	if val[client.TemplateFieldProjectCatalogID] != nil {
+		prjCatID, ok := val[client.TemplateFieldProjectCatalogID].(string)
+		if ok {
+			prjCatalogName = prjCatID
+		} else {
+			prjCatalogName = strings.Split(resource.ID, "-")[0]
+		}
```

Comment:
I guess we should not have the else{} logic, since the link created by the else{} part using the resource.ID won't be functional.